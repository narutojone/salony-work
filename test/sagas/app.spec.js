import { expectSaga } from 'redux-saga-test-plan';

import user, { getProductByID, getProduct, getServicesByID } from 'sagas/app';
import { ActionTypes } from 'constants/index';

jest.mock('modules/client', () => ({
  request: () => ({ items: [] }),
}));

describe('app', () => {
  it('should have the expected watchers', done => {
    expectSaga(user)
      .run({ silenceTimeout: true })
      .then(saga => {
        expect(saga).toMatchSnapshot();
        done();
      });
  });

  it('should match the getProductByID saga', () =>
    expectSaga(getProductByID, { payload: { query: 1350 } })
      .put({
        type: ActionTypes.GET_PRODUCTBYID_SUCCESS,
        payload: {
          data: undefined,
        },
      })
      .run()
  );

  it('should match the getProduct saga', () =>
    expectSaga(getProduct, { payload: { query: 15 } })
      .put({
        type: ActionTypes.GET_PRODUCT_SUCCESS,
        payload: {
          data: undefined,
        },
      })
      .run()
  );

  it('should match the getServices saga', () =>
    expectSaga(getServicesByID, { payload: { query: 15 } })
      .put({
        type: ActionTypes.GET_SERVICESBYID_SUCCESS,
        payload: {
          data: undefined,
        },
      })
      .run()
  );
});
