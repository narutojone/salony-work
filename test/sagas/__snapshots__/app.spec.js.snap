// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should have the expected watchers 1`] = `
Object {
  "fork": Array [
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [
          "GET_PRODUCTBYID",
          "@@redux-saga-test-plan/json/function/getProductByID",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeLatest",
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [
          "GET_PRODUCT",
          "@@redux-saga-test-plan/json/function/getProduct",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeLatest",
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [
          "GET_SERVICESBYID",
          "@@redux-saga-test-plan/json/function/getServicesByID",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeLatest",
      },
    },
  ],
  "take": Array [
    Object {
      "@@redux-saga/IO": true,
      "TAKE": Object {
        "pattern": "GET_PRODUCTBYID",
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "TAKE": Object {
        "pattern": "GET_PRODUCT",
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "TAKE": Object {
        "pattern": "GET_SERVICESBYID",
      },
    },
  ],
}
`;
